aws-rds-customdbengineversion/src/main/java/software/amazon/rds/customdbengineversion/ReadHandler.java

```diff
@@ -0,0 +1,54 @@
+package software.amazon.rds.customdbengineversion;
+
+
+import java.util.Optional;
+
+import software.amazon.awssdk.services.rds.RdsClient;
+import software.amazon.awssdk.services.rds.model.DBEngineVersion;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.HandlerErrorCode;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+import software.amazon.rds.common.handler.Commons;
+import software.amazon.rds.common.handler.HandlerConfig;
+
+
+public class ReadHandler extends BaseHandlerStd {
+    public ReadHandler() {
+        this(CUSTOM_ENGINE_VERSION_HANDLER_CONFIG_10H);
+    }
+
+    public ReadHandler(HandlerConfig config) {
+        super(config);
+    }
+
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final ProxyClient<RdsClient> proxyClient,
+            final Logger logger
+    ) {
+        return proxy.initiate("rds::read-custom-db-engine-version", proxyClient, request.getDesiredResourceState(), callbackContext)
+                .translateToServiceRequest(Translator::describeDbEngineVersionsRequest)
+                .backoffDelay(config.getBackoff())
+                .makeServiceCall((describeDbEngineVersionsRequest, proxyInvocation) -> proxyInvocation.injectCredentialsAndInvokeV2(describeDbEngineVersionsRequest, proxyInvocation.client()::describeDBEngineVersions))
+                .handleError((describeRequest, exception, client, resourceModel, ctx) -> Commons.handleException(
+                        ProgressEvent.progress(resourceModel, ctx),
+                        exception,
+                        DEFAULT_CUSTOM_DB_ENGINE_VERSION_ERROR_RULE_SET))
+                .done((describeCustomDbEngineVersionRequest, describeCustomDbEngineVersionResponse, proxyInvocation, model, context) -> {
+                    final Optional<DBEngineVersion> engineVersion = describeCustomDbEngineVersionResponse.dbEngineVersions().stream().findFirst();
+
+                    if (!engineVersion.isPresent()) {
```

Comment:
Question: does this ever happen? I have added such construction in DBClusterEndpoint because describe API was not throwing an exception, and returning an empty list instead. This is the only describe API that I am aware of that behaves this way (it is a bug that was kept for backwards compatibility reasons). Idiomatic behavior of RDS API is to throw.